lib/spack/spack/concretize.py

```diff
@@ -233,9 +239,6 @@ def concretize_architecture(self, spec):
                 setattr(spec.architecture, field, getattr(arch, field))
                 spec_changed = True
 
-        if not spec.architecture.concrete:
```

Comment:
@becker33 is there a reason to remove this check? If `sys_arch` is guaranteed to be concrete then it would make sense to me that this check isn't needed.